test/spec/utils.ts

```diff
@@ -0,0 +1,191 @@
+}
+
+export interface TestGroup {
+  functionName: string;
+  directory: string;
+  testCases: TestCase[];
+}
+
+export interface TestBatchMeta {
+  directory: string;
+  innerBlsFolder?: boolean;
+  namedYamlFiles?: boolean;
+}
+
+export interface TestBatch {
+  directory: string;
+  testGroups: TestGroup[];
+}
+
+function getTestCasesWithDataYaml(testDirectory: string): TestCase[] {
+  const testCases: TestCase[] = [];
+  for (const testCaseName of fs.readdirSync(testDirectory)) {
+    const testCaseDir = path.resolve(testDirectory, testCaseName);
+    const yamlPath = path.resolve(testCaseDir, "data.yaml");
+    if (!fs.existsSync(yamlPath)) {
+      throw new Error(`Missing yaml data for ${testCaseDir}`);
+    }
+    testCases.push({
+      testCaseName,
+      testCaseData: jsYaml.load(fs.readFileSync(yamlPath, "utf8")) as {
+        input: unknown;
+        output: unknown;
+      },
+    });
+  }
+  return testCases;
+}
+
+function getTestCasesWithNamedYaml(testDirectory: string): TestCase[] {
+  const testCases: TestCase[] = [];
+  for (const testCaseYaml of fs.readdirSync(testDirectory)) {
+    const [testCaseName] = testCaseYaml.split(".");
+    const yamlPath = path.resolve(testDirectory, testCaseYaml);
+    if (!fs.existsSync(yamlPath)) {
+      throw new Error(`Missing yaml data for ${testCaseName}`);
+    }
+    testCases.push({
+      testCaseName,
+      testCaseData: jsYaml.load(fs.readFileSync(yamlPath, "utf8")) as {
+        input: unknown;
+        output: unknown;
+      },
+    });
+  }
+  return testCases;
+}
+
+export function getTestBatch({directory, innerBlsFolder, namedYamlFiles}: TestBatchMeta): TestBatch {
+  const testBatch: TestBatch = {directory, testGroups: []};
+
+  const fullDirPath = path.resolve(REPO_ROOT, directory);
+  for (const functionName of fs.readdirSync(fullDirPath)) {
+    const pathSegments = [fullDirPath, functionName];
+    if (innerBlsFolder) pathSegments.push("bls");
+    const testDirectory = path.resolve(...pathSegments);
+    if (!fs.statSync(testDirectory).isDirectory()) {
+      continue;
+    }
+    const testGroup: TestGroup = {
+      functionName,
+      directory,
+      testCases: namedYamlFiles ? getTestCasesWithNamedYaml(testDirectory) : getTestCasesWithDataYaml(testDirectory),
+    };
+    testBatch.testGroups.push(testGroup);
+  }
+
+  return testBatch;
+}
+
+export function isBlstError(e: unknown): boolean {
+  return (e as Error).message.includes("BLST_ERROR");
+}
+
+const logEmpty = (): void => {};
+
+export type DownloadTestsOptions = {
+  specVersion: string;
+  outputDir: string;
+  /** Root Github URL `https://github.com/ethereum/consensus-spec-tests` */
+  specTestsRepoUrl: string;
+  /** Release files names to download without prefix `["general", "mainnet", "minimal"]` */
+  testsToDownload: string[];
+};
+
+/**
+ * Generic Github release downloader.
+ * Used by spec tests and SlashingProtectionInterchangeTest
+ */
+export async function downloadTests(
```

Comment:
This was pulled out of @lodestar/spec-test-utils but that is an ESM module and has vitest as a sub-dep. `vitest` is ESM only so @lodestar/spec-test-utils cannot be imported